test/geometry/test_stereo.py

```diff
@@ -0,0 +1,198 @@
+    def _create_stereo_camera(batch_size, device, dtype, tx_fx):
+        left_rectified_camera = _SmokeTestData._create_left_camera(batch_size, device, dtype)
+        right_rectified_camera = _SmokeTestData._create_right_camera(batch_size, device, dtype, tx_fx)
+        return left_rectified_camera, right_rectified_camera
+
+
+class TestStereoCamera:
+    """Test class for :class:`~kornia.geometry.camera.stereo.StereoCamera`"""
+
+    @staticmethod
+    def _create_disparity_tensor(batch_size, height, width, max_disparity, device):
+        size = (batch_size, height, width)
+        return torch.randint(size=size, low=0, high=max_disparity, device=device, dtype=torch.float32)
+
+    @staticmethod
+    def test_stereo_camera_attributes_smoke(batch_size, device, dtype):
+        """Test proper setup of the class for smoke data."""
+        tx_fx = -10
+        left_rectified_camera, right_rectified_camera = _SmokeTestData._create_stereo_camera(
+            batch_size, device, dtype, tx_fx
+        )
+
+        stereo_camera = kornia.StereoCamera(left_rectified_camera, right_rectified_camera)
+
+        def _assert_all(x, y):
+            assert torch.all(torch.eq(x, y))
+
+        _assert_all(stereo_camera.fx, _TestParams.fx)
+        _assert_all(stereo_camera.fy, _TestParams.fy)
+        _assert_all(stereo_camera.cx_left, _TestParams.cx)
+        _assert_all(stereo_camera.cy, _TestParams.cy)
+        _assert_all(stereo_camera.tx, -tx_fx / _TestParams.fx)
+
+        assert stereo_camera.Q.shape == (batch_size, 4, 4)
+
+    @staticmethod
+    def test_stereo_camera_attributes_real(batch_size, device, dtype):
+        """Test proper setup of the class for real data."""
+        left_rectified_camera, right_rectified_camera = _RealTestData._get_real_stereo_camera(batch_size, device, dtype)
+
+        stereo_camera = kornia.StereoCamera(left_rectified_camera, right_rectified_camera)
+        assert_allclose(stereo_camera.fx, left_rectified_camera[..., 0, 0])
+        assert_allclose(stereo_camera.fy, left_rectified_camera[..., 1, 1])
+        assert_allclose(stereo_camera.cx_left, left_rectified_camera[..., 0, 2])
+        assert_allclose(stereo_camera.cy, left_rectified_camera[..., 1, 2])
+        assert_allclose(stereo_camera.tx, -right_rectified_camera[..., 0, 3] / right_rectified_camera[..., 0, 0])
+
+        assert stereo_camera.Q.shape == (batch_size, 4, 4)
+
+    def test_reproject_disparity_to_3D_smoke(self, batch_size, device, dtype):
+        """Test reprojecting of disparity to 3D for smoke data."""
+        tx_fx = -10
+        left_rectified_camera, right_rectified_camera = _SmokeTestData._create_stereo_camera(
+            batch_size, device, dtype, tx_fx
+        )
+        disparity_tensor = self._create_disparity_tensor(
+            batch_size, _TestParams.height, _TestParams.width, max_disparity=2, device=device
+        )
+        stereo_camera = kornia.StereoCamera(left_rectified_camera, right_rectified_camera)
```

Comment:
I don't think this should be accessible from the top level api

Reply:
Agree.